src/Tribe/Views/V2/Widgets/List_Widget.php

```diff
@@ -0,0 +1,169 @@
+<?php
+/**
+ * List Widget
+ *
+ * @package Tribe\Events\Views\V2\Widgets
+ * @since   TBD
+ */
+namespace Tribe\Events\Views\V2\Widgets;
+
+use Tribe\Events\Pro\Views\V2\Assets as Pro_Assets;
+use Tribe\Events\Pro\Views\V2\Shortcodes\Tribe_Events;
+use Tribe\Events\Views\V2\Assets as Event_Assets;
+use Tribe\Events\Views\V2\Manager as Views_Manager;
+use Tribe\Events\Views\V2\Theme_Compatibility;
+use Tribe\Events\Views\V2\View;
+use Tribe\Events\Views\V2\View_Interface;
+use Tribe\Widget\Widget_Abstract;
+use Tribe__Context as Context;
+use Tribe__Events__Main as TEC;
+use Tribe__Utils__Array as Arr;
+
+/**
+ * Class for List Widget.
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Views\V2\Widgets
+ */
+class List_Widget extends Widget_Abstract {
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected $slug = 'tribe_events_list_widget';
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected $default_arguments = [
+		// General widget properties
+		'title'         => '',
+
+		// View options
+		'view'                 => null,
+		'should_manage_url'    => false,
+
+		// Event widget options
+		'id'                   => null,
+		'alias-slugs'          => null,
+		'limit'                => 5,
+		'no_upcoming_events'   => '',
+		'featured_events_only' => false,
+		'tribe_is_list_widget' => true,
+		'jsonld_enable'        => true,
+
+		// WP_Widget properties
+		'id_base'              => 'tribe-events-list-widget',
+		'name'                 => null,
+		'widget_options'       => [
+			'classname'   => 'tribe-events-list-widget',
+			'description' => null
+		],
+		'control_options'      => [
+			'id_base' => 'tribe-events-list-widget'
+		],
+	];
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected $validate_arguments_map = [
+		'should_manage_url'    => 'tribe_is_truthy',
+		'no_upcoming_events'   => 'tribe_is_truthy',
+		'featured_events_only' => 'tribe_is_truthy',
+		'jsonld_enable'        => 'tribe_is_truthy',
+	];
+
+	public function get_arguments() {
+		$arguments = $this->default_arguments;
+
+		$arguments['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
+		$arguments['name'   ]         = esc_html__( 'Events List V2', 'the-events-calendar' );
+		$arguments['widget_options']['description'] = esc_html__( 'A widget that displays upcoming events.', 'the-events-calendar' );
+
+		return $this->filter_arguments( $arguments );
+	}
+
+	/**
+	 * Alters the shortcode context with its arguments.
+	 *
+	 * @since  TBD
+	 *
+	 * @param \Tribe__Context $context Context we will use to build the view.
+	 *
+	 * @return \Tribe__Context Context after shortcodes changes.
+	 */
+	public function alter_context( Context $context, array $arguments = [] ) {
+
+		$alter_context = $this->args_to_context( $arguments, $context );
+
+		// The View will consume this information on initial state.
+		$alter_context['id']        = $this->slug;
+
```

Comment:
:warning: Equals sign not aligned correctly; expected 1 space but found 8 spaces

<sub>hash: 8f56658fbd84210202544a70eedd6e84</sub>